File: samples/app-sso/nodejs/server/models/simpleGraphClient.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

const { Client } = require('@microsoft/microsoft-graph-client');
require('isomorphic-fetch');

/**
 * This class is a wrapper for the Microsoft Graph API.
 * See: https://developer.microsoft.com/en-us/graph for more information.
 */
class SimpleGraphClient {
    constructor(token) {
        if (!token || !token.trim()) {
            throw new Error('SimpleGraphClient: Invalid token received.');
        }

        this._token = token;

        // Get an Authenticated Microsoft Graph client using the token issued to the user.
        this.graphClient = Client.init({
            authProvider: (done) => {
                done(null, this._token); // First parameter takes an error if you can't get an access token.
            }
        });
    }

    /**
     *
     * @param {string} searchQuery text to search the inbox for.
     */
    async searchMailInbox(searchQuery) {
        // Searches the user's mail Inbox using the Microsoft Graph API
        return await this.graphClient
            .api('me/mailfolders/inbox/messages')
            .search(searchQuery)
            .get();
    }

    async getMyProfile() {
        try {
            return await this.graphClient.api('/me').get();
        } catch (error) {
            return {};
        }
    }

    // Gets the user's photo
    async getPhotoAsync() {
        const graphPhotoEndpoint = 'https://graph.microsoft.com/v1.0/me/photo/$value';
        const graphRequestParams = {
            method: 'GET',
            headers: {
                'Content-Type': 'image/png',
                authorization: 'bearer ' + this._token
            }
        };

        // eslint-disable-next-line no-undef
        const response = await fetch(graphPhotoEndpoint, graphRequestParams).catch(this.unhandledFetchError);
        if (!response || !response.ok) {
            console.error('User Image Not Found!!');
            // Return a Sample Image
            return 'https://adaptivecards.io/content/cats/1.png';
        }
        const imageBuffer = await response.arrayBuffer().catch(this.unhandledFetchError); // Get image data as raw binary data
        // Convert binary data to an image URL and set the url in state
        const imageUri = 'data:image/png;base64,' + Buffer.from(imageBuffer).toString('base64');
        return imageUri;
    }
}

exports.SimpleGraphClient = SimpleGraphClient;

File: samples/TeamsSDK/Archived/bot-archive-groupchat-messages/python/simple_graph_client.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

import aiohttp
import json
from typing import Dict, Any


class SimpleGraphClient:
    """
    This class is a wrapper for the Microsoft Graph API.
    See: https://developer.microsoft.com/en-us/graph for more information.
    """

    def __init__(self, token: str):
        if not token or not token.strip():
            raise ValueError("SimpleGraphClient: Invalid token received.")
        
        self._token = token
        self._base_url = "https://graph.microsoft.com"

    async def get_messages(self, chat_id: str) -> Dict[str, Any]:
        """
        Gets messages from a specific chat using Microsoft Graph API.
        
        Args:
            chat_id (str): The ID of the chat to fetch messages from
            
        Returns:
            Dict[str, Any]: The response containing chat messages
        """
        url = f"{self._base_url}/beta/chats/{chat_id}/messages"
        
        headers = {
            "Authorization": f"Bearer {self._token}",
            "Content-Type": "application/json"
        }

        async with aiohttp.ClientSession() as session:
            async with session.get(url, headers=headers) as response:
                if response.status == 200:
                    return await response.json()
                else:
                    response_text = await response.text()
                    raise Exception(f"Error fetching messages: {response.status} - {response_text}")
